typescript/sdk/src/providers/MultiProvider.ts

```diff
@@ -277,6 +277,18 @@ export class MultiProvider<MetaExt = {}> extends ChainMetadataManager<MetaExt> {
     return null;
   }
 
+  /**
+   * Get the latest block range for a given chain's RPC provider
+   */
+  async getLatestBlockRange(
+    chainNameOrId: ChainNameOrId,
+    rangeSize = this.getMaxBlockRange(chainNameOrId),
+  ): Promise<{ fromBlock: number; toBlock: number }> {
+    const toBlock = await this.getProvider(chainNameOrId).getBlock('latest');
```

Comment:
I'm slightly concerned that this strategy of establishing a block range with specific block numbers could cause us to miss useful data in the latest block. E.g. if this is called, then some slow other call is made, then the block range is used. By then, the 'latest' we got isn't the real latest. Not a blocking issue, just a concern

Reply:
fair point, though what is an alternative? I think most RPC providers will vomit if you give a from block but to block as latest with "range too big"

Reply:
not sure how I can fix this. Leaving this one out.